### PR TITLE
 Fixes routes on sites in subfolders

### LINKED ISSUE #23: Problem (and solution) for sites in subdirectory
Hi,

Great plugin!

I found an issue when using it in a website that is not in the root folder of the web server. All the links to the endpoints in the documentation would point to `/api-docs/...`. So if I was visiting the docs in `localhost/my-site/api-docs`, for example, all links would point to `localhost/api-docs`.

It took me a while even to notice that, because as all the interaction happens in the React app, the links actually worked. The contents were correctly loaded, even though the address in the address bar would change to an unexisting one.

But after I fixed it, I realized that all the content from the default pages were not being loaded.

So after tweaking a bit, I came up with the solution below.

In `Restsplain.php`, line 73, where you read:

```PHP
	$config = array(
		'basename'          => get_docs_base(),
```

I changed to:

```PHP
	$base_url = parse_url( home_url() );

	$config = array(
		'basename'          => $base_url['path'] . get_docs_base(),
```

I really don't know the implications of that, Ive being playing with the plugin for an hour or so, but it seems to work fine.

If you had the time to have a look and see if it looks good it would be great to see this fix incorporated. I'd be happy to send a pull request or feel free to implement it directly.

cheers

### PR DESCRIPTION
**Pull request by leogermani**
Fixes #23

## Testing instructions

* Test the plugin in a normal installation (not in a subfolder) and confirm routes are still working
* Create an installation in a subfolder of your webserver and confirm routes are still working. 

Check #23 for more details

### CODE REVIEW

File: Restsplain.php [hunk 1]
```diff
@@ -38,11 +38,13 @@ function get_default_docs_base() {
  * @return string
  */
 function get_docs_base() {
+	$subdir_install = (string) parse_url( home_url(), PHP_URL_PATH );
```
Comment (roborourke) [inline Restsplain.php:41]: I think we need to make sure there's no trailing slash here - there generally shouldn't be using `home_url()` without the path but it's a filterable function so best to be certain.

```suggestion
	$subdir_install = rtrim( (string) parse_url( home_url(), PHP_URL_PATH ), '/' );
```

File: Restsplain.php [hunk 2]
```diff
@@ -38,11 +38,13 @@ function get_default_docs_base() {
...
+
 	if ( get_query_var( 'restsplain', false ) ) {
-		return get_default_docs_base();
+		return $subdir_install . get_default_docs_base();
```
Comment (roborourke) [inline Restsplain.php:44]: Realised that `get_default_docs_base()` should probably check it starts with a slash 🤔 Don't worry about that for this PR, just a note to myself.

File: Restsplain.php [hunk 3]
```diff
@@ -38,11 +38,13 @@ function get_default_docs_base() {
...
 	}
 
-	return str_replace( home_url( '/' ), '', get_permalink() );
+	return $subdir_install . str_replace( home_url( '/' ), '', get_permalink() );
```
Comment (roborourke) [inline Restsplain.php:47]: If we know for sure that `$subdir_install` won't have a trailing slash then we should make sure the right hand part of this return value handles that e.g.

```suggestion
	return $subdir_install . '/' . ltrim( str_replace( home_url( '/' ), '', get_permalink() ), '/' );
```

**Review by roborourke — COMMENTED** [review]
Made a couple of notes - the slash handling can be improved a bit because I didn't do a great job of it the first time around!

File: Restsplain.php (hunk 3 above)
Comment (leogermani) [reply Restsplain.php]: Don't we need the same thing on the previous `return` on line 44 then?

Comment (roborourke) [reply Restsplain.php]: Yeah that’s a good point. Although the default value has a leading slash it isn’t guaranteed. Can you do that?

Comment (leogermani) [reply Restsplain.php]: done

**Review by roborourke — APPROVED** [review]
Nice one 😄

**Pull request merged**